dali/core/nvtx.cc

```diff
@@ -56,6 +58,23 @@ DLL_PUBLIC DomainTimeRange::~DomainTimeRange() {
   DomainTimeRangeImpl::GetInstance().Stop();
 }
 
+
+DLL_PUBLIC void SetThreadName(const char *name) {
+  nvtxNameOsThreadA(syscall(SYS_gettid), name);
+  char tmp_name[16];
+  int i = 0;
+  while (name[i] != '\0' && i < 15) {
+    tmp_name[i] = name[i];
+    ++i;
+  }
+  tmp_name[i] = '\0';
+  pthread_setname_np(pthread_self(), tmp_name);
+}
+
+#else
+
+void SetThreadName(const char *name) {}
```

Comment:
The impl without NVTX enabled still could set the pthread name, right?

Reply:
Done